src/qt/metadexdialog.cpp

```diff
@@ -596,13 +617,31 @@ void MetaDExDialog::sendTrade(bool sell)
     if(sell) {
         amountSell = StrToInt64(ui->sellAmountLE->text().toStdString(),divisible);
         price = StrToInt64(ui->sellPriceLE->text().toStdString(),true);
-        if(divisible) { amountDes = (amountSell * price)/COIN; } else { amountDes = amountSell * price; }
+        uint256 amountSell256 = ConvertTo256(amountSell);
+        uint256 priceSell256 = ConvertTo256(price);
+        uint256 totalPrice256 = amountSell256 * priceSell256;
+        if (divisible) totalPrice256 = totalPrice256 / COIN;
+        if (totalPrice256 > std::numeric_limits<int64_t>::max()) {
+            QMessageBox::critical( this, "Unable to send MetaDEx trade",
+            "The total amount for the entered trade details would exceed the maximum amount of tokens." );
+            return;
+        }
+        amountDes = ConvertTo64(totalPrice256);
         if(testeco) { propertyIdDes = 2; } else { propertyIdDes = 1; }
         propertyIdSell = global_metadex_market;
     } else {
         amountDes = StrToInt64(ui->buyAmountLE->text().toStdString(),divisible);
         price = StrToInt64(ui->buyPriceLE->text().toStdString(),true);
-        if(divisible) { amountSell = (amountDes * price)/COIN; } else { amountSell = amountDes * price; }
+        uint256 amountDes256 = ConvertTo256(amountDes);
+        uint256 priceDes256 = ConvertTo256(price);
+        uint256 totalPrice256 = amountDes256 * priceDes256;
+        if (divisible) totalPrice256 = totalPrice256 / COIN;
+        if (totalPrice256 > std::numeric_limits<int64_t>::max()) {
```

Comment:
Sure, thanks for the pointer :)  I've fixed up the other two and repushed the commit.